lib/adapters/class-adapter.php

```diff
@@ -594,6 +594,17 @@ protected function parse_aggregations( array $aggregations ): void {
 		}
 	}
 
+	/**
+	 * Query Elasticsearch directly. Must be implemented in child classes for specific adapters.
+	 *
+	 * @param array $es_args       Arguments to pass to the Elasticsearch server.
+	 *
+	 * @return array|\WP_Error The response from the Elasticsearch server.
+	 */
+	public function search( array $es_args ) {
+		return new \WP_Error( 'not_implemented', 'Method `search()` must be implemented by the child class' );
```

Comment:
I am curious if there's a better way to ensure that all child classes extend this method (since this isn't an abstract class, I can't use an abstract method). I figure either throw an exception or return a WP_Error would be the two best routes.

Reply:
Yeah, I'd throw, personally, but I agree.

Reply:
It looks like it is an abstract class

Reply:
(actually looks at the class file) yep, you're right. Let's make this an abstract method that adapters need to implement.

Reply:
😌  I'm glad I pointed this out, I could have sworn it wasn't abstract! thanks for catching that @mboynes!